src/responsive-navbar/responsive-navbar.component.jsx

```diff
@@ -7,101 +7,134 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import ReactDOM from 'react-dom';
 import Select from 'react-select';
-import { Navbar,
-         Nav,
-         NavItem } from 'react-bootstrap';
-
 import 'react-select/dist/react-select.css';
 import './responsive-navbar.scss';
 
-
 export class ResponsiveNavbar extends React.Component {
 
   constructor(props) {
     super(props);
     this.state = {
-      wrapping: false,
+      updateDimenssions: true,
+      lastVisibleItemIndex: -1,
       lastWidth: 0,
     };
   }
 
   componentDidMount() {
-    window.addEventListener('resize', this.updateDimensions);
+    window.addEventListener('resize', this.handleResizeEvent);
+    window.addEventListener('orientationchange', this.handleResizeEvent); // for mobile support
     // Component is not rendered yet by browser when DidMount is called
     setTimeout(() => {
-      this.updateDimensions();
+      this.handleResizeEvent();
     }, 200);
   }
 
+  componentDidUpdate() {
+    if (this.state.updateDimenssions) {
+      this.setState({ // eslint-disable-line react/no-did-update-set-state
+                      // 2nd render is triggered here in purpose
+        updateDimenssions: false,
+        lastVisibleItemIndex: this.indexOfLastVisibleNavItem(),
+      });
+    }
+  }
+
   componentWillUnmount() {
-    window.removeEventListener('resize', this.updateDimensions);
+    window.removeEventListener('resize', this.handleResizeEvent);
+    window.removeEventListener('orientationchange', this.handleResizeEvent); // for mobile support
   }
 
-  updateDimensions = () => {
-    const firstRef = this.refs[`navitemref${String(0)}`];
-    const lastRef = this.refs[`navitemref${String(this.props.list.length - 1)}`];
-
-    const firstOffsetTop = ReactDOM.findDOMNode(firstRef)
-    ? ReactDOM.findDOMNode(firstRef).offsetTop
-    : 0;
-    const lastOffsetTop = ReactDOM.findDOMNode(lastRef)
-    ? ReactDOM.findDOMNode(lastRef).offsetTop
-    : 0;
-    // Re-render Navbar to see if it fits if screen width increases
-    // Do this once every 50 pixes.
+  indexOfLastVisibleNavItem = () => {
+    const NAV_ITEM_MIN_WIDTH = 200;
```

Comment:
Yes, this was the requirement in the original requirement. The nav item names can be quite long if they are user generated (i.e. several hundred characters and the navbar gets messy with extremely long nav items, the solution was to limit width and truncate the text with text-overflow: ellipsis)